### PR TITLE
[REVIEW] Minor tweaks in .gitignore

### PR DESCRIPTION
**Pull request by eyalroz**
This is for convenience-of-build, only.

* Now ignoring `build/` just like `cpp/build` in case we don't build within the subdirectory
* Comment tweaks

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,21 +13,24 @@ __pycache__
...
 
-##build
+## Python build directories & artifacts
 htmlcov
 dist/
 cudf.egg-info/
 python/build
 python/cudf/bindings/*.cpp
+build/
+cpp/build/
```
Comment (harrism) [inline .gitignore]: This is not a python build directory or artifact. New heading?

Comment (harrism) [reply .gitignore]: Oh I see, it's repeated below. I think you can remove this line.

**Comment by harrism** [comment]
Every PR must now update the CHANGELOG.md with a line starting with the PR# and explaining what the PR introduced. This should go under "Improvements". Once that's done CI will pass for this PR and we can merge.

File: .gitignore (hunk 1 above)
Comment (eyalroz) [reply .gitignore]: Ok. Will do that and update the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,11 +13,15 @@
...
 
-
 ## Bug Fixes
+
 - PR #473 Added missing <random> include
 - PR #495 Updated README to correct where cffi pytest should be executed.
 
+## Non-functional changes
+
+- PR #513 `.gitignore` tweaks
```
Comment (kkraus14) [inline CHANGELOG.md]: Let's move this to `Improvements` and not add another section to the changelog.

Comment (kkraus14) [reply CHANGELOG.md]: @eyalroz yes, this is a code quality improvement and can fit under the `Improvements` section.

Comment (eyalroz) [reply CHANGELOG.md]: Ok, done. Thanks for tolerating my nitpicks.

**Comment by kkraus14** [comment]
Will merge pending CI, thanks @eyalroz!

**Comment by kkraus14** [comment]
Overriding merge because the CI issues are unrelated.

**Pull request merged**